apps/files/tests/test_models.py

```diff
@@ -806,6 +806,40 @@ def test_limit_validator_errors(self):
         assert validation['messages'][0]['type'] == 'error'
 
 
+def test_file_upload_passed_auto_validation_passed():
+    upload = FileUpload(validation=json.dumps({
+        'passed_auto_validation': True,
+    }))
+    assert upload.passed_auto_validation
+
+
+def test_file_upload_passed_auto_validation_failed():
+    upload = FileUpload(validation=json.dumps({
+        'passed_auto_validation': False,
+    }))
+    assert not upload.passed_auto_validation
+
+
+@pytest.mark.parametrize('valid,auto_valid,automated_signing,expected', [
+    (True, True, True, True),
+    (True, False, True, False),
+    (False, True, True, False),
+    (False, False, True, False),
+    (True, True, False, True),
+    (True, False, False, True),
+    (False, True, False, False),
+    (False, False, False, False),
```

Comment:
I added a comment and reordered this a bit. I'm not crazy about it but I think it's more readable than having 8 tests that are almost identical.